### PR TITLE
ab facs: copy histogram functionality

### LINKED ISSUE #655: FACS: instructor can copy histogram from one condition to another
As an instructor, I may want to use the same histogram for multiple combinations of samples and conditions. I'd like to have an easy way to copy the same histogram from other conditions. 

There's a design for this, but it uses a checkbox to initiate the action of choosing conditions to copy. This should be a button, labelled copy instead of a checkbox.

### LINKED ISSUE #667: AB FACS: disable edit and copy buttons for samples without histograms
As an instructor, I don't expect to be able to edit or copy empty histograms.

### PR DESCRIPTION
**Pull request by annagav**
#### What are the relevant tickets?

Fix #655 
Partially fixes #667 
#### What's this PR do?

Add functionality to copy a histogram from my sample to others.

### CODE REVIEW

File: instructor/common.py [hunk 1]
```diff
@@ -1670,6 +1670,40 @@ def facs_histograms_edit(request, assignment, sample_prep, sp):
...
+        mapping = get_object_or_404(
+            models.FlowCytometryHistogramMapping,
+            id=sample['id']
+        )
+        if sample['cell_treatment'] == 'Live':
+            mapping.live_data = histogram
+        elif sample['cell_treatment'] == 'Fixed':
+            mapping.fixed_data = histogram
+        mapping.save()
+    return HttpResponse('complete')
```
Comment (noisecapella) [inline instructor/common.py]: Can you remove `'complete'` here?

File: instructor/common.py [hunk 2]
```diff
@@ -1670,6 +1670,40 @@ def facs_histograms_edit(request, assignment, sample_prep, sp):
...
+@check_assignment_owner
+@login_required
+def copy_histogram(request):
+    """
+    Copy a histogram from one mapping to a given list mappings
+    """
+    mapping_id = request.POST.get('copy_from_pk')
+    cell_treatment = request.POST.get('cell_treatment')
+    copy_to_list = json.loads(request.POST.get('copy_to_list'))
+    copy_from_mapping = get_object_or_404(
```
Comment (noisecapella) [inline instructor/common.py:1735]: We should also filter by assignment id here to make sure the user doesn't access data outside their assignment

File: instructor/common.py [hunk 3]
```diff
@@ -1670,6 +1670,40 @@ def facs_histograms_edit(request, assignment, sample_prep, sp):
...
+
+    if cell_treatment == 'Live':
+        histogram = copy_from_mapping.live_data
+    elif cell_treatment == 'Fixed':
+        histogram = copy_from_mapping.fixed_data
+
+    for sample in copy_to_list:
+        mapping = get_object_or_404(
+            models.FlowCytometryHistogramMapping,
+            id=sample['id']
```
Comment (noisecapella) [inline instructor/common.py]: We should filter by assignment id here

**Comment by noisecapella** [comment]
The issue says that the checkbox should be a button saying 'Copy' instead

**Comment by noisecapella** [comment]
Functionality looks good!

**Comment by annagav** [comment]
Rebased and addressed comments.

File: instructor/common.py (hunk 1 above)
Comment (annagav) [inline instructor/common.py]: done

File: instructor/common.py (hunk 2 above)
Comment (annagav) [inline instructor/common.py:1735]: adding filter

**Comment by noisecapella** [comment]
:+1:

**Pull request merged**